### PR TITLE
Add sklearn 1.5 to CI matrix

### PR DESCRIPTION
**Pull request by Alexsandruss**
### Description

Changes:
- Add sklearn 1.5 (py312) to CI matrix
- Change py38->py39 in CI matrix
- Disable dpctl and dpnp for py39 sklearn 1.0 job due to process hanging during tests
- Deselect two PCA tests related to "auto" solver check
- Update LogReg parameter
- Fix for global patch

Checks:
- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.
- [x] I have resolved any merge conflicts that might occur with the base branch.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [x] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

**Review by icfaust — APPROVED** [review]
Approval contingent on a follow-up ticket for the threading controller issue.

**Comment by ethanglaser** [comment]
There are 386 CI test_patching fails on "tests on GPU" internal CI check for both pythons, this may not quite have been ready for merge
